### PR TITLE
Allow coreJS Symbol to be type object

### LINKED ISSUE #9755: TypeError: Object is not async iterable
I'm having a problem with

@babel/runtime-corejs2/helpers/asyncIterator.js (@7.4.2)

I does not recognize that certain items are iterable because it contains more or less the following:
```js
const Symbol = require("../core-js/symbol");

if (typeof Symbol === "function") {
  // ...
}
```

This is unfortunately incorrect. Symbol as exported by core-js 2 is an object not a function.

It seems like the code should simply allow object or function as the typeof symbol. In this case I am running in a browser with native Symbol support, and `require("../core-js/symbol").iterator === global.Symbol.iterator`, which is the necessary condition for this code to work properly.

### PR DESCRIPTION
**Pull request by conartist6**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #9755
| Patch: Bug Fix?          | Yes
| Major: Breaking Change?  | No
| Minor: New Feature?      | No
| Tests Added + Pass?      | No
| Documentation PR Link    | None
| Any Dependency Changes?  | No
| License                  | MIT

See the #9755 for a detailed description of the problem.

### CODE REVIEW

File: packages/babel-helpers/src/helpers.js [hunk 1]
```diff
@@ -78,9 +78,10 @@ helpers.jsx = helper("7.0.0-beta.0")`
 `;
 
 helpers.asyncIterator = helper("7.0.0-beta.0")`
+  const symbolTypes = ["function", "object"];
   export default function _asyncIterator(iterable) {
     var method
-    if (typeof Symbol === "function") {
+    if (symbolTypes.includes(typeof Symbol)) {
```
Comment (nicolo-ribaudo) [inline packages/babel-helpers/src/helpers.js]: Maybe this could just be `typeof Symbol !== "undefined"`?

Comment (conartist6) [reply packages/babel-helpers/src/helpers.js]: Sounds good to me

**Comment by nicolo-ribaudo** [comment]
@zloirock Is it expected that `core-js` creates an object-type `Symbol` when polyfilling only wks?

**Comment by zloirock** [comment]
@nicolo-ribaudo yes. However, I'm not sure how it's related to `runtime` since here used `Symbol` directly.

**Comment by conartist6** [comment]
This code appears to use Symbol directly, but it is transpiled prior to shipping such that it ends up importing Symbol from core-js.

**Comment by nicolo-ribaudo** [comment]
Someone could also manually import the polyfill for `Symbol.asyncIterable` without importing `Symbol()` polyfill.

**Pull request merged**